lib/vbt/accounts/token.ex

```diff
@@ -134,6 +134,9 @@ defmodule VBT.Accounts.Token do
     end
   end
 
+  defp account_id_field_name(account),
+    do: account.__meta__.schema.__schema__(:association, :tokens).related_key
```

Comment:
![](https://media.giphy.com/media/10CU1Cbz2k2Jpe/giphy.gif)

Reply:
It's actually all in ecto documentation, but scattered over a couple of places. I'll add a comment to clarify it, but the gist is that given an `account` (which is a schema instance), we automatically deduce the field name of its `:tokens` association (e.g. `account_id`).

We start with `__meta__` (publicly documented in Schema.t) to get the schema meta, and from there the schema module (`schema`). The `__schema__(:association, :tokens)` reflection mechanism is mentioned [here](https://hexdocs.pm/ecto/Ecto.Schema.html#module-reflection), and then we have to pick up the related_key (e.g. `account_id`), mentioned [here](https://hexdocs.pm/ecto/Ecto.Association.Has.html).